nni/retiarii/nn/pytorch/component.py

```diff
@@ -69,7 +68,7 @@ def _replicate_and_instantiate(blocks, repeat):
         assert repeat <= len(blocks), f'Not enough blocks to be used. {repeat} expected, only found {len(blocks)}.'
         blocks = blocks[:repeat]
         if not isinstance(blocks[0], nn.Module):
-            blocks = [b() for b in blocks]
+            blocks = [b(i) for i, b in enumerate(blocks)]
```

Comment:
how can you make sure `i` is the input argument of a block?

Reply:
The function is provided by user, and user should make sure of it.
Yes, this is an API change. But I think this should not be a problem because the old version doesn't work at all, and no one actually complained about it.

Reply:
so, the document has not been updated in that pr, right? i cannot find the explanation of how to write a block

Reply:
The guide to use APIs has already been rewritten in #4399.

Originally, I think these changes should also go with #4399. But I split them into this PR because I consider them as bug fixes, thus should be merged into this release.